Cargo.toml

```diff
@@ -34,21 +34,23 @@ tower = { version = "0.4", optional = true }
 axum = { version = "0.7.4", optional = true }
 pin-project-lite = { version = "0.2", optional = true }
 futures-util = { version = "0.3", default_features = false, features = [], optional = true }
-hyper = { version = "1.1.0", default-features = false, features = ["http1", "client"], optional = true }
+hyper = { version = "0.14", default-features = false, features = ["http1", "client"], optional = true }
+hyper-v1 = { package = "hyper", version = "1.1", default-features = false, features = ["http1", "client"], optional = true }
```

Comment:
`opentelemetry-http` uses hyper@0.14 and it's not really compatible with the latest one.

And we need hyper@1 for out test setup, but this one is hidden behind `test` feature.

Reply:
Perhaps we should add an `http` feature and hide `opentelemetry-http` and `hyper` behind it.

But it will complicate protocol detection quite a lot since we'll have to insert a lot of `cfg` macros to handle cases when http support is disabled.

Reply:
I get the sense that it will take a while for hyper 1.x support https://github.com/open-telemetry/opentelemetry-rust/issues/1427

Reply:
Yeah, looks like it.

I even tried to implement my own `HyperClient` for the latest hyper version, but it didn't work since new hyper uses http@1, but `opentelemetry-http` is written on top of hyper@0.14 and http@0.x.